src/iuse.cpp

```diff
@@ -7851,22 +7851,23 @@ int iuse::washclothes( player *p, item *it, bool, const tripoint& )
         return 0;
     }
 
-    const inventory &crafting_inv = p->crafting_inventory();
-    if( !crafting_inv.has_charges( "water", 40 ) && !crafting_inv.has_charges( "water_clean", 40 ) ) {
-        p->add_msg_if_player( _( "You need a large amount of fresh water to use this." ) );
-        return 0;
-    }
-
     const int pos = g->inv_for_flag( "FILTHY", _( "Wash what?" ) );
     item &mod = p->i_at( pos );
     if( pos == INT_MIN ) {
         p->add_msg_if_player( m_info, _( "Never mind." ) );
         return 0;
     }
+    
+    const int required_water = 2 * mod.volume() / 250_ml;
+    const inventory &crafting_inv = p->crafting_inventory();
+    if( !crafting_inv.has_charges( "water", required_water ) && !crafting_inv.has_charges( "water_clean", required_water ) ) {
+        p->add_msg_if_player( _( "You need %1$i charges of water to wash your %2$s." ), required_water, mod.tname().c_str() );
```

Comment:
Slightly misleading? Above checks for both X `water` and X `water_clean`, whereas this line only mentions the former.